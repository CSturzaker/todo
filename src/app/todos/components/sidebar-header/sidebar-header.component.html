<img
  mat-card-avatar
  src="https://avatars2.githubusercontent.com/u/465825?s=200&v=4"
  alt="avatar-img"
/>
{{ userName }}
