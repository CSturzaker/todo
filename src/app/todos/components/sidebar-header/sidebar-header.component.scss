:host {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 20px 0 20px 10px;
  font-weight: 500;
  img {
    margin-right: 10px;
  }
}
