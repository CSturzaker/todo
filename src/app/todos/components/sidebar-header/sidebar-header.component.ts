import { Component, OnInit, Input } from '@angular/core'

@Component({
  selector: 'todo-sidebar-header',
  templateUrl: './sidebar-header.component.html',
  styleUrls: ['./sidebar-header.component.scss'],
})
export class SidebarHeaderComponent implements OnInit {
  @Input()
  userName: string
  constructor() {}

  ngOnInit() {}
}
