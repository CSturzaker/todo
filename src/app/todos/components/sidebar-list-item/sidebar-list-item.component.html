<p>sidebar-list-item works!</p>
