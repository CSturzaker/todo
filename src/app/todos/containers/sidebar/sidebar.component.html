<todo-sidebar-header [userName]="userName$ | async"></todo-sidebar-header>
<div class="nav-list">
  <todo-sidebar-list-item
    *ngFor="let item of navItems"
    [item]="item"
  ></todo-sidebar-list-item>
</div>
