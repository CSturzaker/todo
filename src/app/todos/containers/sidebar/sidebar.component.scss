.nav-list {
  width: 100%;
}
