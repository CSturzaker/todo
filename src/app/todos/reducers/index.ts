import {
  createSelector,
  createFeatureSelector,
  Action,
  combineReducers,
} from '@ngrx/store'
import * as fromRoot from '../../state'
import * as fromTodos from '../reducers/todo.reducer'

export const todoFeatureKey = 'todo'

export interface TodoState {
  [fromTodos.todoFeatureKey]: fromTodos.State
}

export interface State extends fromRoot.State {
  [todoFeatureKey]: TodoState
}

export function reducers(state: TodoState | undefined, action: Action) {
  return combineReducers({
    [fromTodos.todoFeatureKey]: fromTodos.reducer,
  })(state, action)
}

export const selectTodoState = createFeatureSelector<State, TodoState>(
  todoFeatureKey
)

export const selectTodoStatusState = createSelector(
  selectTodoState,
  (state: TodoState) => state.todo
)

export const selectTodos = createSelector(
  selectTodoStatusState,
  fromTodos.getTodoItems
)
