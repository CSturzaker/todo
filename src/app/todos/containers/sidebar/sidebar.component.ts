import { Component, OnInit } from '@angular/core'
import { Store, select } from '@ngrx/store'
import { Observable } from 'rxjs'

import * as fromAuth from '../../../auth/reducers'
import { ListItem } from '../../models'

@Component({
  selector: 'todo-sidebar',
  templateUrl: './sidebar.component.html',
  styleUrls: ['./sidebar.component.scss'],
})
export class SidebarComponent implements OnInit {
  userName$: Observable<string> = this.store.pipe(
    select(fromAuth.selectUserName)
  )

  navItems: ListItem[] = [
    {
      label: 'Team To-Do List',
      icon: 'list',
    },
  ]

  constructor(private store: Store<fromAuth.State>) {}

  ngOnInit() {}
}
