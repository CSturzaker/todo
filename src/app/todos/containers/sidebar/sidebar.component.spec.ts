import { ComponentFixture, TestBed } from '@angular/core/testing'
import { provideMockStore, MockStore } from '@ngrx/store/testing'
import { Store } from '@ngrx/store'
import { MatIconModule } from '@angular/material'

import { SidebarComponent } from './sidebar.component'
import { SidebarHeaderComponent } from '../../components/sidebar-header/sidebar-header.component'
import { SidebarListItemComponent } from '../../components/sidebar-list-item/sidebar-list-item.component'

import * as fromAuth from '../../../auth/reducers'

describe('SidebarComponent', () => {
  let component: SidebarComponent
  let fixture: ComponentFixture<SidebarComponent>
  let store: MockStore<fromAuth.State>

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [MatIconModule],
      declarations: [
        SidebarComponent,
        SidebarHeaderComponent,
        SidebarListItemComponent,
      ],
      providers: [
        provideMockStore({
          selectors: [{ selector: fromAuth.selectUserName, value: 'test' }],
        }),
      ],
    })
    fixture = TestBed.createComponent(SidebarComponent)
    component = fixture.componentInstance

    store = TestBed.get(Store)

    spyOn(store, 'dispatch')
  })

  it('should create', () => {
    fixture.detectChanges()
    expect(component).toMatchSnapshot()
  })
})
