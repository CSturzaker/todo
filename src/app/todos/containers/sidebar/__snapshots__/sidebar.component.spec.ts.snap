// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should create 1`] = `
SidebarComponent {
  "navItems": Array [
    Object {
      "icon": "list",
      "label": "Team To-Do List",
    },
  ],
  "store": MockStore {
    "_isScalar": false,
    "actionsObserver": ActionsSubject {
      "_isScalar": false,
      "_value": Object {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "dispatch": [Function],
    "initialState": Object {},
    "lastState": Object {},
    "reducerManager": MockReducerManager {
      "_isScalar": false,
      "_value": [Function],
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "scannedActions$": Observable {
      "_isScalar": false,
      "source": ActionsSubject {
        "_isScalar": false,
        "_value": Object {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
    },
    "source": MockState {
      "_isScalar": false,
      "_value": Object {},
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [
        MapSubscriber {
          "_parent": DistinctUntilChangedSubscriber {
            "_parent": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_parents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "hasKey": true,
            "isStopped": false,
            "key": "test",
            "keySelector": undefined,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "_parents": null,
          "_subscriptions": Array [
            SubjectSubscription {
              "_parent": [Circular],
              "_parents": null,
              "_subscriptions": null,
              "closed": false,
              "subject": [Circular],
              "subscriber": [Circular],
            },
          ],
          "closed": false,
          "count": 1,
          "destination": DistinctUntilChangedSubscriber {
            "_parent": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_parents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "hasKey": true,
            "isStopped": false,
            "key": "test",
            "keySelector": undefined,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "isStopped": false,
          "project": [Function],
          "syncErrorThrowable": true,
          "syncErrorThrown": false,
          "syncErrorValue": null,
          "thisArg": [Circular],
        },
      ],
      "thrownError": null,
    },
    "state$": MockState {
      "_isScalar": false,
      "_value": Object {},
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [
        MapSubscriber {
          "_parent": DistinctUntilChangedSubscriber {
            "_parent": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_parents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "hasKey": true,
            "isStopped": false,
            "key": "test",
            "keySelector": undefined,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "_parents": null,
          "_subscriptions": Array [
            SubjectSubscription {
              "_parent": [Circular],
              "_parents": null,
              "_subscriptions": null,
              "closed": false,
              "subject": [Circular],
              "subscriber": [Circular],
            },
          ],
          "closed": false,
          "count": 1,
          "destination": DistinctUntilChangedSubscriber {
            "_parent": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_parents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": Subscriber {
              "_parent": null,
              "_parents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parent": null,
                "_parentSubscriber": [Circular],
                "_parents": null,
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "hasKey": true,
            "isStopped": false,
            "key": "test",
            "keySelector": undefined,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "isStopped": false,
          "project": [Function],
          "syncErrorThrowable": true,
          "syncErrorThrown": false,
          "syncErrorValue": null,
          "thisArg": [Circular],
        },
      ],
      "thrownError": null,
    },
  },
  "userName$": Store {
    "_isScalar": false,
    "actionsObserver": ActionsSubject {
      "_isScalar": false,
      "_value": Object {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "operator": DistinctUntilChangedOperator {
      "compare": undefined,
      "keySelector": undefined,
    },
    "reducerManager": MockReducerManager {
      "_isScalar": false,
      "_value": [Function],
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "source": Store {
      "_isScalar": false,
      "actionsObserver": ActionsSubject {
        "_isScalar": false,
        "_value": Object {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "operator": MapOperator {
        "project": [Function],
        "thisArg": undefined,
      },
      "reducerManager": MockReducerManager {
        "_isScalar": false,
        "_value": [Function],
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "source": MockStore {
        "_isScalar": false,
        "actionsObserver": ActionsSubject {
          "_isScalar": false,
          "_value": Object {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [],
          "thrownError": null,
        },
        "dispatch": [Function],
        "initialState": Object {},
        "lastState": Object {},
        "reducerManager": MockReducerManager {
          "_isScalar": false,
          "_value": [Function],
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "_isScalar": false,
          "source": ActionsSubject {
            "_isScalar": false,
            "_value": Object {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "hasError": false,
            "isStopped": false,
            "observers": Array [],
            "thrownError": null,
          },
        },
        "source": MockState {
          "_isScalar": false,
          "_value": Object {},
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [
            MapSubscriber {
              "_parent": DistinctUntilChangedSubscriber {
                "_parent": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "_parents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "hasKey": true,
                "isStopped": false,
                "key": "test",
                "keySelector": undefined,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "_parents": null,
              "_subscriptions": Array [
                SubjectSubscription {
                  "_parent": [Circular],
                  "_parents": null,
                  "_subscriptions": null,
                  "closed": false,
                  "subject": [Circular],
                  "subscriber": [Circular],
                },
              ],
              "closed": false,
              "count": 1,
              "destination": DistinctUntilChangedSubscriber {
                "_parent": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "_parents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "hasKey": true,
                "isStopped": false,
                "key": "test",
                "keySelector": undefined,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "project": [Function],
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
              "thisArg": [Circular],
            },
          ],
          "thrownError": null,
        },
        "state$": MockState {
          "_isScalar": false,
          "_value": Object {},
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [
            MapSubscriber {
              "_parent": DistinctUntilChangedSubscriber {
                "_parent": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "_parents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "hasKey": true,
                "isStopped": false,
                "key": "test",
                "keySelector": undefined,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "_parents": null,
              "_subscriptions": Array [
                SubjectSubscription {
                  "_parent": [Circular],
                  "_parents": null,
                  "_subscriptions": null,
                  "closed": false,
                  "subject": [Circular],
                  "subscriber": [Circular],
                },
              ],
              "closed": false,
              "count": 1,
              "destination": DistinctUntilChangedSubscriber {
                "_parent": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "_parents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": Subscriber {
                  "_parent": null,
                  "_parents": null,
                  "_subscriptions": Array [
                    [Circular],
                  ],
                  "closed": false,
                  "destination": SafeSubscriber {
                    "_complete": undefined,
                    "_context": Object {
                      "unsubscribe": [Function],
                    },
                    "_error": [Function],
                    "_next": [Function],
                    "_parent": null,
                    "_parentSubscriber": [Circular],
                    "_parents": null,
                    "_subscriptions": null,
                    "closed": false,
                    "destination": Object {
                      "closed": true,
                      "complete": [Function],
                      "error": [Function],
                      "next": [Function],
                    },
                    "isStopped": false,
                    "syncErrorThrowable": false,
                    "syncErrorThrown": false,
                    "syncErrorValue": null,
                  },
                  "isStopped": false,
                  "syncErrorThrowable": true,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "hasKey": true,
                "isStopped": false,
                "key": "test",
                "keySelector": undefined,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "project": [Function],
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
              "thisArg": [Circular],
            },
          ],
          "thrownError": null,
        },
      },
    },
  },
}
`;
